test/token/ERC721/extensions/ERC721Consecutive.test.js

```diff
@@ -0,0 +1,128 @@
+    { receiver: user3, amount: 0 },
+    { receiver: user1, amount: 7 },
+  ];
+
+  beforeEach(async function () {
+    this.token = await ERC721ConsecutiveMock.new(
+      name,
+      symbol,
+      batches.map(({ receiver }) => receiver),
+      batches.map(({ amount }) => amount),
+    );
+  });
+
+  describe('minting during construction', function () {
+    it('events are emitted at construction', async function () {
+      let first = 0;
+
+      for (const batch of batches) {
+        if (batch.amount == 0) continue;
+
+        await expectEvent.inTransaction(this.token.transactionHash, this.token, 'ConsecutiveTransfer', {
+          fromTokenId: web3.utils.toBN(first),
+          toTokenId: web3.utils.toBN(first + batch.amount - 1),
+          fromAddress: constants.ZERO_ADDRESS,
+          toAddress: batch.receiver,
+        });
+
+        first += batch.amount;
+      }
+    });
+
+    it('ownership is set', async function () {
+      const owners = batches.flatMap(({ receiver, amount }) => Array(amount).fill(receiver));
+
+      for (const tokenId in owners) {
+        expect(await this.token.ownerOf(tokenId))
+          .to.be.equal(owners[tokenId]);
+      }
+    });
+
+    it('balance & voting poser are set', async function () {
+      for (const account of accounts) {
+        const balance = batches
+          .filter(({ receiver }) => receiver === account)
+          .map(({ amount }) => amount)
+          .reduce((a, b) => a + b, 0);
+
+        expect(await this.token.balanceOf(account))
+          .to.be.bignumber.equal(web3.utils.toBN(balance));
+
+        expect(await this.token.getVotes(account))
+          .to.be.bignumber.equal(web3.utils.toBN(balance));
+      }
+    });
+
+    it('enumerability correctly set', async function () {
+      const owners = batches.flatMap(({ receiver, amount }) => Array(amount).fill(receiver));
+
+      expect(await this.token.totalSupply())
+        .to.be.bignumber.equal(web3.utils.toBN(owners.length));
+
+      for (const tokenId in owners) {
+        expect(await this.token.tokenByIndex(tokenId))
+          .to.be.bignumber.equal(web3.utils.toBN(tokenId));
+      }
+
+      for (const account of accounts) {
+        const owned = Object.entries(owners)
+          .filter(([ _, owner ]) => owner === account)
+          .map(([ tokenId, _ ]) => tokenId);
+
+        for (const i in owned) {
+          expect(await this.token.tokenOfOwnerByIndex(account, i).then(x => x.toString()))
+            .to.be.bignumber.equal(web3.utils.toBN(owned[i]));
+        }
+      }
+    });
+  });
+
+  describe('minting after construction', function () {
+    it('consecutive minting is not possible after construction', async function () {
+      await expectRevert(
+        this.token.mintConsecutive(user1, 10),
+        'ERC721Consecutive: batch minting restricted to constructor',
+      );
+    });
+
+    it('simple minting is possible after construction', async function () {
```

Comment:
Let's add a test that it's not possible to mint a token id that was batch-minted.